src/bin/driver.rs

```diff
@@ -719,13 +758,24 @@ mod app {
         c.local.driver_dac[channel as usize].set(current).unwrap()
     }
 
-    #[task(priority = 1, shared=[alarm_handle, laser_interlock])]
+    #[task(priority = 1, shared=[alarm_handle, laser_interlock, output_state])]
     fn trip_alarm(mut c: trip_alarm::Context, reason: Reason) {
         c.shared.alarm_handle.lock(|handle| *handle = None);
         log::error!("Alarm! {:?} Laser Interlock tripped.", reason);
-        c.shared
-            .laser_interlock
-            .lock(|ilock| ilock.set(Some(reason)));
+        (c.shared.laser_interlock, c.shared.output_state).lock(
+            |ilock, state| {
+                let delays = ilock.set(Some(reason), state);
+                delays.iter().enumerate().for_each(|(i, delay)| {
+                    if let Some(delay) = delay {
+                        handle_output_tick::spawn_after(
+                            delay.convert(),
+                            Channel::try_from(i).unwrap(),
+                        )
+                        .unwrap();
+                    }
+                });
```

Comment:
This looks like something refactorable (it shows up five times). Or at least the part except the `spawn_after`.